tests/dogtag/pytest-ansible/pytest/performance_test/test_cert_revocation.py

```diff
@@ -0,0 +1,75 @@
+"""
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Description: Load Test client Server Side configuration
+#   Steps :-
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Author: Deepak Punia <dpunia@redhat.com>
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Copyright Red Hat, Inc.
+# SPDX-License-Identifier: GPL-2.0-or-later
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+"""
+from pki.client import PKIConnection
+from pki.cert import CertClient
+import sys
+import logging
+import threading
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--hostname", help="<CA hostname>")
+parser.add_argument("--port", help="<CA Port Number>")
+parser.add_argument("--client-cert", help="path for admin.pem certificate")
+parser.add_argument("--number-of-clients", help="Number of thread", type=int)
+parser.add_argument("--number-of-tests-per-client", help="Number of test per thread", type=int)
+
+log = logging.getLogger()
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+args = parser.parse_args()
+
+# Create a PKIConnection object that stores the details of the CA.
+connection = PKIConnection('https', args.hostname, args.port)
+
+# The pem file used for authentication. Created from a p12 file using the
+# command -
+# openssl pkcs12 -in <p12_file_path> -out /tmp/auth.pem -nodes
+connection.set_authentication_cert(args.client_cert)
+
+# Instantiate the CertClient
+cert_client = CertClient(connection)
+
+
+def run_test(number_of_tests_per_thread):
+    # execute the specified number of tests sequentially
+    # sn > 20 to avoid revocation of initial system certificate
+    for sn in range(20, number_of_tests_per_thread):
```

Comment:
I think here we're assuming that the cert enrollment test was done using sequential serial numbers (instead of random serial numbers), which is the default so that's fine. Assuming all enrollments completed successfully, there should be N certs with serial numbers ranging from `sn_1` to `sn_N`. We need to determine the value of `sn_1` after the enrollment test and provide that value to the revocation test.

If we run the loop above with more than 1 thread it will try to revoke the same certs multiple times (because it uses the same sn range for all threads). I think we need to make sure that each cert is only revoked once with something like this:
```
thread_id = <1, 2, ... number_of_threads>
begin_sn = sn_1 + number_of_tests_per_thread * (thread_id - 1)
end_sn = sn_1 + number_of_tests_per_thread * thread_id - 1

for sn in range(begin_sn, end_sn):
    ...
```

Reply:
Another option is to store the issued serial numbers into files, then reread the files for revocation test. This method will work for both sequential and random serial numbers.

Reply:
Thanks Endi for the comment,
I have tried with the first approach serial numbers ranging but i have found some inconsistency in that while running like begin_sn and end_sn, they are not providing exact expected output and above calculation also giving me same value for both begin_sn and end_sn.

So i have choose the second option to store the issued serial numbers into files and that's applied in revocation script.

Please review further in-case anything need to update.